be/src/exprs/cast_expr.cpp

```diff
@@ -1465,8 +1465,7 @@ Expr* VectorizedCastExprFactory::create_primitive_cast(ObjectPool* pool, const T
         from_type = TYPE_VARCHAR;
     }
     if (from_type == TYPE_NULL) {
-        // NULL TO OTHER TYPE, direct return
-        from_type = to_type;
+        from_type = TYPE_BOOLEAN;
```

Comment:
here, ban casting NULL_TYPE to BOOLEAN, just return error msg as follows:
```
return Status::NotSupported("Not support cast from NULL_TYPE to BOOLEAN");
```
TYPE_NULL should not propagated to BE from FE.